contracts/deployables/erc20/VotesERC20LockableV1.sol

```diff
@@ -77,7 +73,19 @@ contract VotesERC20LockableV1 is ILockableV1, IMintableV1, VotesERC20V1 {
         }
     }
 
+    function setMaxTotalSupply(uint256 newMaxTotalSupply) external onlyOwner {
+        uint256 currentlyMaxTotalSupply = maxTotalSupply;
+        maxTotalSupply = newMaxTotalSupply;
+        if (currentlyMaxTotalSupply != newMaxTotalSupply) {
+            emit MaxTotalSupplyUpdated(newMaxTotalSupply);
+        }
+    }
+
     function mint(address to, uint256 amount) external onlyOwner {
+        uint256 newTotalSupply = totalSupply() + amount;
+        if (newTotalSupply > maxTotalSupply) {
+            revert ExceedMaxTotalSupply();
+        }
         _mint(to, amount);
     }
```

Comment:
Just to be clear, seems like this code allows for setting a new `maxTotalSupply` which is _less than_ the current amount of minted tokens. It would not allow any MORE tokens to be minted, but it does create a strange state.

Is this what we want?

I'm not even sure how we would be able to check the current amount of minted tokens (in order to provide a safety check)...

Reply:
yeah, it didn't check if new `maxTotalSupply` is smaller, I thought it may make sense with future burnable feature so that `totalSupply` can decrease.
can't we used `totalSupply()` to get "current amount of minted tokens"?

Reply:
though you are right, this create a strange state, will update.